CredentialProvider/CredentialProviderPresenter.swift

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import AuthenticationServices
+import SwiftKeychainWrapper
+
+@available(iOS 12, *)
+class CredentialProviderPresenter {
+    weak var view: CredentialProviderViewProtocol?
+    private let profile: Profile
+    
+    init(view: CredentialProviderViewProtocol, profile: Profile = ExtensionProfile(localName: "profile")) {
```

Comment:
nit: If we're going to use the `ExtensionProfile` here could we update the comment here https://github.com/mozilla-mobile/firefox-ios/blob/main/Extensions/NotificationService/ExtensionProfile.swift#L9 to keep things up to date?